controller/src/util/instance_action.rs

```diff
@@ -108,37 +108,43 @@ async fn internal_do_instance_watch(
 async fn handle_instance(
     event: WatchEvent<KubeAkriInstance>,
     kube_interface: &impl KubeInterface,
-) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+) -> Result<(), anyhow::Error> {
     trace!("handle_instance - enter");
     match event {
         WatchEvent::Added(instance) => {
             info!(
                 "handle_instance - added Akri Instance {}: {:?}",
                 instance.metadata.name, instance.spec
             );
-            handle_instance_change(&instance, &InstanceAction::Add, kube_interface).await?;
+            handle_instance_change(&instance, &InstanceAction::Add, kube_interface)
+                .await
+                .map_err(|e| anyhow::format_err!("{}", e))?;
             Ok(())
         }
         WatchEvent::Deleted(instance) => {
             info!(
                 "handle_instance - deleted Akri Instance {}: {:?}",
                 instance.metadata.name, instance.spec
             );
-            handle_instance_change(&instance, &InstanceAction::Remove, kube_interface).await?;
+            handle_instance_change(&instance, &InstanceAction::Remove, kube_interface)
+                .await
+                .map_err(|e| anyhow::format_err!("{}", e))?;
             Ok(())
         }
         WatchEvent::Modified(instance) => {
             info!(
                 "handle_instance - modified Akri Instance {}: {:?}",
                 instance.metadata.name, instance.spec
             );
-            handle_instance_change(&instance, &InstanceAction::Update, kube_interface).await?;
-            Ok(())
-        }
-        WatchEvent::Error(ref e) => {
-            trace!("handle_instance - error for Akri Instance: {}", e);
+            handle_instance_change(&instance, &InstanceAction::Update, kube_interface)
+                .await
+                .map_err(|e| anyhow::format_err!("{}", e))?;
             Ok(())
         }
+        WatchEvent::Error(ref e) => Err(anyhow::format_err!(
```

Comment:
nit - How about adding a comment on why bubbling up the error is needed here? or is this a test only scenario?

Reply:
I'm not sure if we even want to bubble this up. I am using this PR to test the e2e tests to figure out why they are failing.